app/controllers/submissions_controller.rb

```diff
@@ -9,6 +9,7 @@ def create
 
     @task = Task.find(params[:submission][:task_id])
     @task.update status: :waiting if @task.not_submitted? || @task.waiting?
+    @task.update status: :new_version if @task.accepted_partially?
```

Comment:
Maybe change names to something like `new_submission`, `new_submission_with_notes` or `on_review`, `on_review_with_notes`? Things like `accepted_partially` look odd to me now as well. Do you have any ideas on in?